Use gawk regardless of what was picked up during build.

diff -u -r ../texinfo-6.5/texindex/texindex.in ./texindex/texindex.in
--- ../texinfo-6.5/texindex/texindex.in	2016-02-06 12:21:52.000000000 +0000
+++ ./texindex/texindex.in	2018-07-12 07:29:20.427030482 +0000
@@ -44,7 +44,7 @@
 #
 # else use configured value for awk.
 if test -z "$awk_binary"; then
-  awk_binary="@AWK@"
+  awk_binary=gawk
 fi
 #
 # that should never be empty, but just in case, else fall back to plain
